src/charm.py

```diff
@@ -147,7 +140,7 @@ def get_resource(self) -> Optional[str]:
             logger.debug("cannot fetch charm resource")
             return None
 
-        if not os.path.getsize(snap_path) > 0:
+        if not os.path.getsize(snap_path):
```

Comment:
This is logically weird to understand, although it has the same effect. `os.path.getsize` return number, and `not <number>` is not logic (although it makes sense in Python)

Reply:
Or we can change it to:

```python
# No double negative, and size > 0 return boolean.
if os.path.getsize(snap_path) > 0:
    return snap_path
else:
    logger.debug("resource is an empty file")
    return None
```

(Or any recommend?)

Reply:
I'm happy either way.  I do agree that retaining a comparison with an integer makes it easier to read the condition.  A third option if we don't want to reorder the code and want to avoid the double negative:

```python
if os.path.getsize(snap_path) == 0:
    .... return None

...
```

Reply:
I change it to

```
if os.path.getsize(snap_path) <= 0
```

to avoid double negative.